File: Leetcode/res/Word Break/4.py
# Author: allannozomu
# Runtime: 40 ms
# Memory: 13.9 MB

class Solution:
    
    def word_break_aux(self, s: str, wordDict: List[str], dp) -> bool:
        if (len(s) == 0):
            return True
        
        if s in dp:
            return dp[s]
        
        for word in wordDict:
            if word == s[0:len(word)]:
                if self.word_break_aux(s[len(word):], wordDict, dp):
                    return True
                
        dp[s] = False
        return False
    
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        dp = {}
        return self.word_break_aux(s, wordDict, dp)

File: Leetcode/res/Single Number/4.cpp
\*
Author: allannozomu
Runtime: 12 ms
Memory: 9.8 MB*\

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int i = 0;
        for (int num : nums){
            i ^= num;
        }
        return i;
    }
};